src/Umbraco.Web/Routing/ContentFinderByUrlAlias.cs

```diff
@@ -90,9 +93,20 @@ bool IsMatch(IPublishedContent c, string a1, string a2)
                 {
                     v = c.Value<string>(propertyAlias);
                 }
+
                 if (string.IsNullOrWhiteSpace(v)) return false;
-                v = "," + v.Replace(" ", "") + ",";
-                return v.InvariantContains(a1) || v.InvariantContains(a2);
+
+                // Build up a comparison list of the current node's urlAlias to compare with the current request
+                var sb = new StringBuilder(",");
+                foreach (var urlAlias in v.Split(','))
+                {
+                    // Add the node's urlAlias to the list
+                    sb.Append(urlAlias.Replace(" ", "") + ",");
+                    // Add the root node's absolute url joined to the node's urlAlias to the list
+                    sb.Append(rootAbsoluteUrl + (urlAlias.StartsWith("/") || rootAbsoluteUrl.EndsWith("/") ? "" : "/") + urlAlias.Replace(" ", "") + ",");
```

Comment:
If the `rootAbsoluteUrl` ends with a slash AND the `urlAlias` starts with a slash, you'll not match here because there will be a double slash.  Better to add `.TrimEnd('/') + '/'` to then end of lines 118 and 124 below (to ensure the `rootAbsoluteUrl` does end in a slash), and change this line to read as follows:
```
sb.Append(rootAbsoluteUrl + urlAlias.TrimStart('/').Replace(" ", "") + ",");
```

Reply:
There's an extension method within Umbraco for doing exactly this: https://github.com/umbraco/Umbraco-CMS/blob/7f1d5fe7cc4615846f2df555246bb58e8cd023c8/src/Umbraco.Core/StringExtensions.cs#L463-L466

Reply:
@benjaminc Good shout! I'll change that asap! 👍 
@ronaldbarendse Ah awesome! Thanks for letting me know about that 😀